scheduler/scheduler.go

```diff
@@ -149,9 +150,20 @@ func BuildDependencyGraph(c client.Interface, sel labels.Selector) (*DependencyG
 			depGraph[d.Parent].RequiredBy, depGraph[d.Child])
 	}
 
-	return &depGraph, nil
+	cycles := DetectCycles(depGraph)
```

Comment:
I'd move all this cycles part to main(), between building and executing depgraph. Now BuildDependencyGraph func is already too big'n'complex. I'm thinking how to refactor it a bit.